interactions/client.py

```diff
@@ -389,11 +405,44 @@ def decorator(coro: Coroutine) -> Callable[..., Any]:
                 raise InteractionException(
                     11, message="Your command needs at least one argument to return context."
                 )
-            if options is not MISSING and len(coro.__code__.co_varnames) + 1 < len(options):
-                raise InteractionException(
-                    11,
-                    message="You must have the same amount of arguments as the options of the command.",
-                )
+            if options is not MISSING:
+                if len(coro.__code__.co_varnames) + 1 < len(options):
+                    raise InteractionException(
+                        11,
+                        message="You must have the same amount of arguments as the options of the command.",
+                    )
+                if isinstance(options, List) and len(options) > 25:
+                    raise InteractionException(
+                        11, message="You cannot have more than 25 options in a command"
+                    )
+                _option: Option
+                for _option in options:
+                    if _option.type not in (
+                        OptionType.SUB_COMMAND,
+                        OptionType.SUB_COMMAND_GROUP,
+                    ):
+                        if getattr(_option, "autocomplete", False) and getattr(
+                            _option, "choices", False
+                        ):
+                            raise InteractionException(
+                                11,
+                                message="Autocomplete may not be set to true if choices are present",
```

Comment:
This sounds like you raise an exception as an information, I'd do an `log.warning` instead if there's no need to cancel the process.

Reply:
Yes, but according to the API, this sounds illegal. If I just throw a warning and let this pass to API requests, will this cause errors down the line?

Reply:
I don't know, but the usage of "may" implies to me, that it *could* be the case but it *does not have to* what, imo, should not raise an exception then.

Reply:
Alright, that's fair. Will change